components/service/experiments/src/main/java/mozilla/components/service/experiments/Experiments.kt

```diff
@@ -47,34 +48,39 @@ open class ExperimentsInternalAPI internal constructor() {
      * @param applicationContext [Context] to access application features, such
      * as shared preferences.
      */
-    fun initialize(
+    open fun initialize(
         applicationContext: Context,
-        fetchClient: Client = HttpURLConnectionClient()
+        configuration: Configuration = Configuration()
     ) {
         if (isInitialized) {
             logger.error("Experiments library should not be initialized multiple times")
             return
         }
 
+        // Any code below might trigger recording into Glean, so make sure Glean is initialized.
+        if (!getGlean().isInitialized()) {
```

Comment:
Do we want to really stop init in this case? Glean doesn't crash if a recording is attempted before init is complete: we'll simply discard, if it's *too* early. Maybe printing the error is enough?

Reply:
Not doing this would make Glean an optional dependency.
What i want to make sure is that all experimentation data is recorded, to make semantics clear.

This could technically initialize here, but i want to enforce Glean as a strict dependency to reduce potential for errors.

Reply:
I see now, makes sense then!